core/pool.go

```diff
@@ -9,7 +9,7 @@ import (
 type Pool struct {
 }
 
-func (p Pool) Start(engines []Engine, provider RequestProvider, metricsManager MetricsManager, config ChukonuConfig) {
+func (p Pool) Start(engines []Engine, provider RequestProvider, metricsManager MetricsManager, config ChukonuConfig, fuse chan bool, ack chan bool) {
```

Comment:
Yes, let's defer it for now. For now, let's assume a reasonable threshold should be able to achieve a reasonable balancing point, like setting the threshold at 75%, and knowing it could go above but not too much, ending at like 90%. 
We should know more once we start evaluating